lib/checkbufferoverrun.cpp

```diff
@@ -1803,12 +1803,23 @@ void CheckBufferOverrun::checkInsecureCmdLineArgs()
             const Token* tok = function->token;
 
             // Get the name of the argv variable
-            unsigned int varid = 0;
-            if (Token::Match(tok, "main ( int %var% , char * %var% [ ] ,|)")) {
-                varid = tok->tokAt(7)->varId();
-
-            } else if (Token::Match(tok, "main ( int %var% , char * * %var% ,|)")) {
-                varid = tok->tokAt(8)->varId();
+            unsigned int argvVarid = 0;
+            if (Token::simpleMatch(tok, "main ("))
+                tok = tok->tokAt(2);
+            else
+                continue;
+            if (Token::Match(tok, "int %var% , char * %var% [ ] ,|)")) {
```

Comment:
I suggest that you skip first argument here. Like:

    if (Token::Match(tok, "const| int %var% ,"))
        tok = tok->nextArgument();
    else
        continue;